crates/solidity/inputs/language/bindings/rules.msgb

```diff
@@ -1060,3 +1065,253 @@ attribute symbol_reference = symbol  => type = "push_symbol", symbol = symbol, i
+  edge @path.lexical_scope -> @var_assign.lexical_scope
+}
+
+@var_assign [YulVariableAssignmentStatement @expr expression: [YulExpression]] {
+  edge @expr.lexical_scope -> @var_assign.lexical_scope
+}
+
+;;; Function definitions
+
+@block [YulBlock [YulStatements [YulStatement @fundef [YulFunctionDefinition]]]] {
+  ;; Function definitions are hoisted in the enclosing block
+  edge @block.fundefs -> @fundef.def
+  ;; The only definitions available in the function's lexical scope (other than
+  ;; parameters) are functions (ie. the body of the function doesn't have access
+  ;; to any outside variables)
+  edge @fundef.lexical_scope -> @block.fundefs
+}
+
+@fundef [YulFunctionDefinition
+    @name name: [YulIdentifier]
+    @body body: [YulBlock]
+] {
+  node @fundef.lexical_scope
+  node @fundef.def
+
+  node def
+  attr (def) node_definition = @name
+  attr (def) definiens_node = @fundef
+
+  edge @fundef.def -> def
+  edge @body.lexical_scope -> @fundef.lexical_scope
+}
+
+@fundef [YulFunctionDefinition [YulParametersDeclaration [YulParameters
+    @param [YulIdentifier]
+]]] {
+  node def
+  attr (def) node_definition = @param
+  attr (def) definiens_node = @param
+
+  edge @fundef.lexical_scope -> def
+}
+
+@fundef [YulFunctionDefinition [YulReturnsDeclaration [YulVariableNames
+    @return_param [YulIdentifier]
+]]] {
+  node def
+  attr (def) node_definition = @return_param
+  attr (def) definiens_node = @return_param
+
+  edge @fundef.lexical_scope -> def
+}
+
+;;; (Deprecated) Stack assignment
+
+@stmt [YulStatement [YulStackAssignmentStatement @name [YulIdentifier]]] {
+  node ref
+  attr (ref) node_reference = @name
+
+  edge ref -> @stmt.lexical_scope
+}
+
+;;; If statements
+
+@stmt [YulStatement [YulIfStatement
+    @condition condition: [YulExpression]
+    @body body: [YulBlock]
+]] {
+  edge @condition.lexical_scope -> @stmt.lexical_scope
+  edge @body.lexical_scope -> @stmt.lexical_scope
+  edge @body.fundefs -> @stmt.fun_scope
+}
+
+;;; Switch statements
+
+@stmt [YulStatement [YulSwitchStatement
+    @expr expression: [YulExpression]
+]] {
+  edge @expr.lexical_scope -> @stmt.lexical_scope
+}
+
+@stmt [YulStatement [YulSwitchStatement [YulSwitchCases [YulSwitchCase
+    [_ @body body: [YulBlock]]
+]]]] {
+  edge @body.lexical_scope -> @stmt.lexical_scope
+  edge @body.fundefs -> @stmt.fun_scope
+}
+
+;;; For statements
+
+@stmt [YulStatement [YulForStatement
+    @init initialization: [YulBlock]
+    @cond condition: [YulExpression]
+    @iter iterator: [YulBlock]
+    @body body: [YulBlock]
+]] {
+  edge @init.lexical_scope -> @stmt.lexical_scope
+  edge @cond.lexical_scope -> @stmt.lexical_scope
+  edge @cond.lexical_scope -> @init.defs
+  edge @iter.lexical_scope -> @stmt.lexical_scope
+  edge @iter.lexical_scope -> @init.defs
+  edge @body.lexical_scope -> @stmt.lexical_scope
+  edge @body.lexical_scope -> @init.defs
+}
+
+;;; (Deprecated) Label statements
```

Comment:
nit: I suggest not having `Deprecated` here, since we officially support all language versions, and I want to keep the distinction between things deprecated/removed from a specific Solidity version, and things we "slang" deprecated in our own API/grammar/tools.

Reply:
Yes, you're right. I may add some conditionals using `version-matches` instead to show that these are rules that apply only to some deprecated versions of Solidity.